MetovaTestKitTests/CellTestingTests/MTKCollectionViewTestCellTests.swift

```diff
@@ -0,0 +1,115 @@
+//  distribute, sublicense, and/or sell copies of the Software, and to
+//  permit persons to whom the Software is furnished to do so, subject to
+//  the following conditions:
+//
+//  The above copyright notice and this permission notice shall be
+//  included in all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+//  LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+//  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+//  WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import XCTest
+
+@testable import MetovaTestKit
+
+class CollectionViewTestCellTests: MTKBaseTestCase {
+    
+    func testGoodCell() {
+        let dataSource = CollectionDataSource(items: 1)
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+        let indexPath = IndexPath(item: 0, section: 0)
+        
+        collectionView.dataSource = dataSource
+        
+        let testExecutedExpectation = expectation(description: "Test block should be executed!")
+        collectionView.testCell(at: indexPath) { (testCell: MockCollectionCellTypeA) in
+            testExecutedExpectation.fulfill()
+        }
+        wait(for: [testExecutedExpectation], timeout: 0)
+    }
+    
+    func testNoDataSource() {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+        let indexPath = IndexPath(item: 0, section: 0)
+        
+        let description = "failed - \(collectionView) does not have a dataSource"
+        expectTestFailure(TestFailureExpectation(description: description, lineNumber: #line+1)) {
+            collectionView.testCell(at: indexPath) { (testCell: UICollectionViewCell) in
+                XCTFail("Test closure should not be called when cell can not be properly instantiated and cast.")
+            }
+        }
+    }
+    
+    func testNoCellFailure() {
+        let dataSource = CollectionDataSource()
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+        let indexPath = IndexPath(item: 0, section: 0)
+        
+        collectionView.dataSource = dataSource
+
+        let description = "failed - \(collectionView) does not have a cell at \(indexPath)"
+        expectTestFailure(TestFailureExpectation(description: description, lineNumber: #line+1)) {
+            collectionView.testCell(at: indexPath) { (testCell: UICollectionViewCell) in
+                XCTFail("Test closure should not be called when cell can not be properly instantiated and cast.")
+            }
+        }
+    }
+    
+    func testWrongCellTypeFailure() {
```

Comment:
Perhaps expand name to `testAssertionFailureDueToIncorrectCellType` or something similarly expressive.